hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -701,6 +701,7 @@ object HoodieSparkSqlWriter {
     val bootstrapBasePath = hoodieConfig.getStringOrThrow(BASE_PATH,
       s"'${BASE_PATH.key}' is required for '${BOOTSTRAP_OPERATION_OPT_VAL}'" +
         " operation'")
+    assert(!path.equals(bootstrapBasePath), "Bootstrap base path and path must be different")
```

Comment:
```suggestion
    assert(!path.equals(bootstrapBasePath), "Bootstrap source base path and Hudi table base path must be different")
```

Reply:
Could you also add the same assert for other bootstrap flow, including Deltastreamer, Hudi CLI, and Spark SQL Procedure?